sycl/doc/extensions/SRGB/srgb_support.asciidoc

```diff
@@ -0,0 +1,163 @@
+
+
+== Overview
+
+sRGB isn't an image data format, per se. It is a linearization of RGB color values that adjusts the color to be better matched with a particular hardware medium. A more detailed explanation may be found here: https://software.intel.com/content/www/us/en/develop/articles/using-opencl-20-srgb-image-format.html
+
+OpenCL provides automatic linearization of RGB values into sRGB color space. OneAPI now exposes that OpenCL support for read_only images. 
+
+== New ImageChannelOrder Enums
+
+sRGB values have been added to the `image_channel_order` enumeration. The `image_channel_order::srgba` value allows images to be read with sRGB linearization of the color values. This value can be passed to `image` constructors, with some limitations that are outlined below.
+
+[source]
+----
+image_channel_order::srgba
+----
+
+== Limitations of sRGB usage
+
+There are five primary limitations of which any potential user should be aware.
+
+. 2D and 3D images only. 
+. unorm_int8 channel type (only)
+. read_only access to image 
+. const data required when initializing image.
+. OpenCL backend only at this time.
+
+Three of these are limitations to the construction of the image object itself, and will typically result in a `CL_IMAGE_FORMAT_NOT_SUPPORTED` error if exceeded. 
+
+=== 2D and 3D images only.
+The title is self explanatory. OneAPI does not support 1D sRGB images. They must be 2D or 3D images.
+
+=== unorm_int8 channel type (only)
+sRGB images can *only* be created with the `image_channel_type::unorm_int8` channel type. This is an 8-bit per channel type, normalized to a low accuracy floating point on read. 
+
+=== read_only access to image
+The OneAPI support for sRGB images at this time is read_only. Be sure to only request read access mode accessors, not write or read_write.
+
+=== const data required when initializing image
+As a consequence of the read_only limitation, the image must be initialized with user data upon construction, and that data must be `const`.
+
+=== OpenCL backend only at this time
+The sRGB enablement is available on OpenCL, but no other backends at this time.   
+
+== Example
+
+[source]
+----
+using accessorPixelT = sycl::float4; // we access the image data as four floats
+using dataPixelT = uint32_t;         // but the underlying data is merely 32 bit per pixel (four 8 bit values)
+
+// this should yield a read of approximate 0.5 for each channel
+// when read directly with a normal non-linearized image (e.g. image_channel_order::rgba).  
+// For sRGB (image_channel_order::srgba), this is the value that will experience maximum conversion/linearization. 
+// So we should read values of approximately 0.2 
+dataPixelT basicPixel{127 << 24 | 127 << 16 | 127 << 8 | 127};
+
+const sycl::range<2> ImgRange_2D(width, height);
+
+// Make a 2D image with every pixel initialized to basicPixel.
+// IMPORTANT: const data is *required* for sRGBA images.
+// OpenCL support is limited for 2D/3D images that are read only.
```

Comment:
But other backends may support RW images, right? So, why to have a hard limitation in the spec?

Reply:
No other backends support sRGB images presently.  This isn't a full or final specification of how sRGB support should be in SYCL. That's the work we'll be seeing in future SYCL spec updates, and image support is changing considerably. This is an extension doc for the sRGB support we are able to add in today.  I wanted this example in the docs to be usable and to guide the user to working code. The const data limitation and using the OpenCL backend are real requirements.

Reply:
OK